### PR TITLE
Fix brew completion for `brew install`

### PR DESCRIPTION
**Pull request by tjjh89017**
## Description

Since brew mirgate Formula into Taps diectory
Change the way to find out all Homebrew-Core Formula
Compatible with old version Homebrew

### CODE REVIEW

File: share/completions/brew.fish [hunk 1]
```diff
@@ -17,9 +17,9 @@ function __fish_brew_using_command
 end
 
 function __fish_brew_formulae
-    set -l formuladir (brew --repository)/Library/Formula/
+    set -l formuladir (brew --repository)/Library/Formula/ (brew --repository)/Library/Taps/homebrew/homebrew-core/Formula
     # __fish_complete_suffix .rb
-    ls $formuladir/*.rb | sed 's/.rb$//' | sed "s|^$formuladir||"
+    find $formuladir -name "*.rb" -type f 2> /dev/null | sed 's/.rb$//' | sed 's|^/\([^/]*/\)*||'
```
Comment (floam) [inline share/completions/brew.fish]: Let's try to use the `string` builtin here instead of `sed`

**Comment by floam** [comment]
Might it assume less and be more future-proof (and able to generalize to other Taps) if you get the Formula directory via: `brew --repository homebrew/homebrew-core`?

**Comment by floam** [comment]
This is pretty broken in master right now, I'm merging the fix.

**Comment by floam** [comment]
Wait, this is wrong:

`set -l formuladir (brew --repository)/Library/Formula/ (brew --repository homebrew/core)`

**Comment by tjjh89017** [comment]
`brew --repository homebrew/core` is working fine.
But now I just use `brew search` to list all local formula, however it has some problem when `tap-pin` is used.

**Comment by floam** [comment]
Something like this will get a working path for the main keg.

```
> echo (brew --repository homebrew/homebrew-core)/Formula
/usr/local/Library/Taps/homebrew/homebrew-core/Formula
```

**Comment by floam** [comment]
Compared to what I was getting from the PR update:
```
> echo (brew --repository)/Library/Formula/ (brew --repository homebrew/core)
/usr/local/Library/Formula/ /usr/local/Library/Taps/homebrew/homebrew-core
> ls /usr/local/Library/Formula/
ls: /usr/local/Library/Formula/: No such file or directory
```

**Comment by floam** [comment]
Oh, I see, probably that's fine and for old installs.

**Comment by tjjh89017** [comment]
`/usr/local/Library/Formula/` for compatible
`find` will search recursively because some custom tap won't have `Formula` directory.
Dont specify `Formula` for try to modify easily to search all formula from Taps.
But it seems useless, while using `brew search` instead

**Comment by floam** [comment]
I will try this soon. How is the speed of `brew search`?

**Comment by tjjh89017** [comment]
`brew search` seem a little slower than `find` & `string replace` using Macbook Air 2014.

**Comment by floam** [comment]
That's much better than I'm getting. Maybe something is wrong with my install. 

```
floam@MacBook-Pro ~> brew search !> /dev/null
floam@MacBook-Pro ~> echo $CMD_DURATION 
14438
```

**Comment by floam** [comment]
I like it, and you can just use `brew search` directly now instead of wrapping it in that function.

**Comment by tjjh89017** [comment]
With fresh install Hackintosh (Yosemite) last week.

```
$ find . -name "*.rb" -type f 2> /dev/null | string replace -r '^/(?:[^/]*/)*([^/]*)\.rb' '$1' 1> /dev/null
$ echo $CMD_DURATION
233
$ brew search 1> /dev/null
$ echo $CMD_DURATION
333
```

**Comment by floam** [comment]
Let me know when you're done and happy and I'll merge this to master.

Thanks!

**Comment by tjjh89017** [comment]
For `brew install` I think it is done.
You could merge it when anytime you want.
I will work on for other issues (e.g. `tap-pin`) and create another PR for fix
Thanks a lot. :)

**Pull request merged**